### PR TITLE
Improve comments, docs, function names in the SIR interpreter.

### PR DESCRIPTION
**Pull request by ptersilie**
This addresses some of the comments in #226.

### CODE REVIEW

File: internal_ws/ykbh/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/// The stopgap interpreter which is invoked during a guard failure. It picks up right after the
```
Comment (ltratt) [inline internal_ws/ykbh/src/lib.rs:1]: Module-level comments need to be `//!`.

**Comment by ltratt** [comment]
Just one small comment.

File: internal_ws/ykbh/src/lib.rs (hunk 1 above)
Comment (ptersilie) [reply internal_ws/ykbh/src/lib.rs]: Fixed in 3961562.

**Comment by ptersilie** [comment]
New commits ready for re-reviewing.

File: internal_ws/ykbh/src/lib.rs [hunk 2]
```diff
@@ -348,7 +351,7 @@ impl SIRInterpreter {
                         _ => unreachable!(),
                     };
                     // Get a pointer to the return value of the called frame.
-                    let ret_ptr = oldframe.mem.local_ptr(&Local(0));
+                    let ret_ptr = oldframe.mem.local_ptr(&RETURN_LOCAL);
```
Comment (ltratt) [inline internal_ws/ykbh/src/lib.rs:354]: There are other instance of `Local(0)` that presumably also need changing -- you'll need to grep the whole repo for these I suspect.

Comment (ptersilie) [reply internal_ws/ykbh/src/lib.rs:354]: Yeah, see https://github.com/softdevteam/yk/pull/226/files#r573834812.

My plan was to do these PRs on a per file basis, but realised now that may not be a good idea.

Comment (ltratt) [reply internal_ws/ykbh/src/lib.rs:354]: I think per-file might not quite work, at least not for this particular case.

Comment (ptersilie) [reply internal_ws/ykbh/src/lib.rs:351]: Fixed in 8f5174b.

**Comment by ptersilie** [comment]
I think this should be it for this PR, if we want to keep them small.

**Comment by ltratt** [comment]
I think there are `Local(0)`'s that aren't covered by https://github.com/softdevteam/yk/pull/234/commits/8f5174b60b3fac60cb8ddcf6366948b730b1a9df (e.g. in `ykcompile.rs`). Can you confirm if you've been able to grep over the repo for `Local.0`? Also I'm not sure about some of the other `Local`s -- greping for `Local.[0-9]` shows up other things like `Local(2)` which may or may not need to be tackled (maybe in another PR)?

**Comment by ptersilie** [comment]
> I think there are `Local(0)`'s that aren't covered by [8f5174b](https://github.com/softdevteam/yk/commit/8f5174b60b3fac60cb8ddcf6366948b730b1a9df) (e.g. in `ykcompile.rs`). Can you confirm if you've been able to grep over the repo for `Local.0`? Also I'm not sure about some of the other `Local`s -- greping for `Local.[0-9]` shows up other things like `Local(2)` which may or may not need to be tackled (maybe in another PR)?

The remaining ones that you see should all be in `tests/src/ykcompile.rs`, where the locals do not refer to anything special, i.e. they are not related to a return local or interpreter context.

**Comment by ltratt** [comment]
Aha! OK, please squash.

**Comment by ptersilie** [comment]
Squashed and rebased.

**Comment by ptersilie** [comment]
bors r-

Forgot to format.

**Comment by ptersilie** [comment]
Fixed. Please try again.

**Comment by ltratt** [comment]
OK, but let's not force push without a reviewer OK in the future please.

bors r+

**Comment by ptersilie** [comment]
Even for commits that only ran `cargo fmt`? Fine.

**Comment by ltratt** [comment]
I know it can seem a bit persnickety, but it removes the element of personal judgment, which makes it easier for hard situations IMHO.

**Pull request merged**